core-tests/src/test/java/org/mule/runtime/core/MuleMessageDataTypePropagationTestCase.java

```diff
@@ -109,15 +107,15 @@ public void updatesDataTypeWhenPayloadIsReplacedWithNullPayload() throws Excepti
     {
         MuleMessage muleMessage = MuleMessage.builder().payload(TEST).build();
 
-        assertDataType(MuleMessage.builder(muleMessage).payload(NullPayload.getInstance()).build(), Object.class, ANY, null);
+        assertDataType(MuleMessage.builder().nullPayload().build(), Object.class, ANY, null);
```

Comment:
why was the cloning removed? if it's not needed anymore, remove the previous line too.